<?xml version="1.0"?>

<launch>
  <!-- Topics. -->
  <arg name="sensor_sub_topic" default="/sensor" />
  <arg name="other_traj_topics" default="[]" />
  <arg name="updated_env_topic" default="/updated_env" />
  <arg name="vis_topic" default="/vis/known_env" />
  <arg name="other_occupancy_grid_topics" default="[]" />

  <!-- Services. -->
  <arg name="other_bound_srvs" default="[]" />
  <arg name="replan_srv" default="/replan" />
  <arg name="bound_srv" default="/bound" />
  <arg name="dynamics_srv" default="/planner_dynamics" />

  <!-- Probabilistic planning params. -->
  <arg name="collision_threshold" default="0.1" />
  <arg name="collision_check_resolution" default="0.33" />
  <arg name="grid_resolution" default="0.33" />

  <!-- Fixed frame. -->
  <arg name="fixed_frame" default="world" />

  <!-- Maximum planning runtime (sec). -->
  <arg name="max_runtime" default="0.5" />

  <!-- State space bounds [x, y, z, vx, vy, vz].
       NOTE! These should agree with the upper and lower environment bounds. -->
  <arg name="state_upper" default="[10.0, 10.0, 10.0, 10.0, 10.0, 10.0]" />
  <arg name="state_lower" default="[-10.0, -10.0, 0.0, -10.0, -10.0, -10.0]" />

  <!-- Environment parameters. -->
  <arg name="env_upper_x" default="10.0" />
  <arg name="env_upper_y" default="10.0" />
  <arg name="env_upper_z" default="10.0" />
  <arg name="env_lower_x" default="10.0" />
  <arg name="env_lower_y" default="10.0" />
  <arg name="env_lower_z" default="0.0" />

  <!-- Planner node.  -->
  <node name="planner"
        pkg="fastpeople"
        type="time_varying_astar_node"
        output="screen">
    <param name="topic/sensor_sub" value="$(arg sensor_sub_topic)" />
    <rosparam param="topic/other_trajs" subst_value="True">$(arg other_traj_topics)</rosparam>
    <rosparam param="topic/other_occupancy_grids" subst_value="True">$(arg other_occupancy_grid_topics)</rosparam>
    <param name="topic/updated_env" value="$(arg updated_env_topic)" />
    <param name="vis/env" value="$(arg vis_topic)" />

    <param name="srv/replan" value="$(arg replan_srv)" />
    <param name="srv/dynamics" value="$(arg dynamics_srv)" />
    <param name="srv/bound" value="$(arg bound_srv)" />
    <rosparam param="srv/other_bounds" subst_value="True">$(arg other_bound_srvs)</rosparam>

    <param name="max_runtime" value="$(arg max_runtime)" />

    <param name="frame/fixed" value="$(arg fixed_frame)" />

    <rosparam param="state/upper" subst_value="True">$(arg state_upper)</rosparam>
    <rosparam param="state/lower" subst_value="True">$(arg state_lower)</rosparam>

    <param name="env/upper/x" value="$(arg env_upper_x)" />
    <param name="env/upper/y" value="$(arg env_upper_y)" />
    <param name="env/upper/z" value="$(arg env_upper_z)" />
    <param name="env/lower/x" value="$(arg env_lower_x)" />
    <param name="env/lower/y" value="$(arg env_lower_y)" />
    <param name="env/lower/z" value="$(arg env_lower_z)" />

    <param name="collision_threshold" value="$(arg collision_threshold)" />
    <param name="collision_check_resolution" value="$(arg collision_check_resolution)" />
    <param name="grid_resolution" value="$(arg grid_resolution)" />
  </node>
</launch>
